kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/Config.java

```diff
@@ -350,56 +366,256 @@ public Config(String masterUrl, String apiVersion, String namespace, boolean tru
-    this.requestConfig.setImpersonateExtras(impersonateExtras);
+    this(masterUrl, apiVersion, namespace, trustCerts, disableHostnameVerification, caCertFile, caCertData,
+        clientCertFile, clientCertData, clientKeyFile, clientKeyData, clientKeyAlgo, clientKeyPassphrase, username,
+        password, oauthToken, autoOAuthToken, watchReconnectInterval, watchReconnectLimit, connectionTimeout, requestTimeout,
+        scaleTimeout, loggingInterval, maxConcurrentRequests, maxConcurrentRequestsPerHost, http2Disable,
+        httpProxy, httpsProxy, noProxy, userAgent, tlsVersions, websocketPingInterval, proxyUsername, proxyPassword,
+        trustStoreFile, trustStorePassphrase, keyStoreFile, keyStorePassphrase, impersonateUsername, impersonateGroups,
+        impersonateExtras, oauthTokenProvider, customHeaders, requestRetryBackoffLimit, requestRetryBackoffInterval,
+        uploadRequestTimeout, onlyHttpWatches, currentContext, contexts, false);
+  }
 
-    this.http2Disable = http2Disable;
-    this.httpProxy = httpProxy;
-    this.httpsProxy = httpsProxy;
-    this.noProxy = noProxy;
-    this.proxyUsername = proxyUsername;
-    this.proxyPassword = proxyPassword;
-    this.userAgent = userAgent;
-    this.tlsVersions = tlsVersions;
-    this.trustStoreFile = trustStoreFile;
-    this.trustStorePassphrase = trustStorePassphrase;
-    this.keyStoreFile = keyStoreFile;
-    this.keyStorePassphrase = keyStorePassphrase;
-    this.oauthTokenProvider = oauthTokenProvider;
-    this.customHeaders = customHeaders;
+  @Buildable(builderPackage = "io.fabric8.kubernetes.api.builder", editableEnabled = false)
+  public Config(String masterUrl, String apiVersion, String namespace, boolean trustCerts, boolean disableHostnameVerification,
+      String caCertFile, String caCertData, String clientCertFile, String clientCertData, String clientKeyFile,
+      String clientKeyData, String clientKeyAlgo, String clientKeyPassphrase, String username, String password,
+      String oauthToken, String autoOAuthToken, int watchReconnectInterval, int watchReconnectLimit, int connectionTimeout,
+      int requestTimeout,
+      long scaleTimeout, int loggingInterval, int maxConcurrentRequests, int maxConcurrentRequestsPerHost,
+      boolean http2Disable, String httpProxy, String httpsProxy, String[] noProxy,
+      String userAgent, TlsVersion[] tlsVersions, long websocketPingInterval, String proxyUsername,
+      String proxyPassword, String trustStoreFile, String trustStorePassphrase, String keyStoreFile, String keyStorePassphrase,
+      String impersonateUsername, String[] impersonateGroups, Map<String, List<String>> impersonateExtras,
+      OAuthTokenProvider oauthTokenProvider, Map<String, String> customHeaders, int requestRetryBackoffLimit,
+      int requestRetryBackoffInterval, int uploadRequestTimeout, boolean onlyHttpWatches, NamedContext currentContext,
+      List<NamedContext> contexts, boolean autoConfigure) {
+    autoConfigure(this, null);
```

Comment:
This looks weird. Why is autoconfigure always invoked? It wasn't before (in some of the methods)